### PR TITLE
Add TensorFlow/Keras Neural Networks for Classification

### PR DESCRIPTION
**Pull request by asRot0**
### Describe your change:

`fully_connected_mnist.py` demonstrates a basic fully-connected feed-forward network with two hidden layers, ReLU activation, and dropout for regularization.

`cnn_mnist.py` implements a Convolutional Neural Network (CNN) with multiple convolutional, pooling, and batch normalization layers for feature extraction and down-sampling, followed by fully connected layers for classification.

* [x] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [x] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [x] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: machine_learning/neural_networks/fully_connected_mnist.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import tensorflow as tf
+from tensorflow.keras import layers, models
+
+
+# Load the MNIST dataset from Keras
+mnist = tf.keras.datasets.mnist
+(X_train, y_train), (X_test, y_test) = mnist.load_data()
+
+# Normalize the images from 0-255 to 0-1 by dividing by 255
+X_train, X_test = X_train / 255.0, X_test / 255.0
```
[the remark this answers is not in the document]
Comment (asRot0) [reply machine_learning/neural_networks/fully_connected_mnist.py]: solved

File: machine_learning/neural_networks/fully_connected_mnist.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import tensorflow as tf
+from tensorflow.keras import layers, models
+import numpy as np
+
+# Load the MNIST dataset from Keras
+mnist = tf.keras.datasets.mnist
+(X_train, y_train), (X_test, y_test) = mnist.load_data()
+
+# Normalize the images from 0-255 to 0-1 by dividing by 255
+X_train, X_test = X_train / 255.0, X_test / 255.0
```
[the remark this answers is not in the document]
Comment (asRot0) [reply machine_learning/neural_networks/fully_connected_mnist.py]: done

File: machine_learning/neural_networks/fully_connected_mnist.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+# (None in this case)
+
+# Third-party imports
+import numpy as np
+import tensorflow as tf
+from tensorflow.keras import layers, models
+
+# Load the MNIST dataset from Keras
+mnist = tf.keras.datasets.mnist
+(X_train, y_train), (X_test, y_test) = mnist.load_data()
```
[the remark this answers is not in the document]
Comment (asRot0) [reply machine_learning/neural_networks/fully_connected_mnist.py]: done

File: machine_learning/neural_networks/fully_connected_mnist.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+"""
+
+import numpy as np
+import tensorflow as tf
+from tensorflow.keras import layers, models
+
+
+# Load the MNIST dataset from Keras
+mnist = tf.keras.datasets.mnist
+(X_train, y_train), (X_test, y_test) = mnist.load_data()
```
[the remark this answers is not in the document]
Comment (asRot0) [reply machine_learning/neural_networks/fully_connected_mnist.py]: done

File: machine_learning/neural_networks/cnn_mnist.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+"""
+
+# import tensorflow as tf
+from tensorflow.keras import layers, models
+from tensorflow.keras.datasets import mnist
+from tensorflow.keras.utils import to_categorical
+
+# Load and preprocess the MNIST data
+(X_train, y_train), (X_test, y_test) = mnist.load_data()
```
[the remark this answers is not in the document]
Comment (asRot0) [reply machine_learning/neural_networks/cnn_mnist.py]: done

**Review by asRot0 — COMMENTED** [review]
done

**Review by imSanko — APPROVED** [review]
Good

**Review by asRot0 — COMMENTED** [review]
done

**Review by imSanko — APPROVED** [review]

**Comment by tianyizheng02** [comment]
Sorry, but we can't accept this PR because it's a how-to.

Your PR just uses imports all of the necessary NN functionality from TF. This repo is for educational purposes, and as such you need to provide your own implementation of your proposed algorithms. You did not implement the NNs yourself, and simply using TF to develop your models doesn't teach readers how a NN is actually implemented.
  
This was all stated in this repo's contributing guidelines, which you indicated that you have read:
  > * [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
  
As stated in the contributing guidelines:
> Algorithms in this repo should not be how-to examples for existing Python packages. Instead, they should perform internal calculations or manipulations to convert input values into different output values. Those calculations or manipulations can use data types, classes, or functions of existing Python packages but each algorithm in this repo should add unique value.
